syntax = "proto3";

message Report {
  uint32 pool_id = 1;
  uint32 session_id = 2;
  uint32 time_stamp = 3;

  // CPU
  message CPU {
    uint32 logical_cores = 1;
    float current_freq = 2;
    float max_freq = 3;
    float percent = 4;
    float load_1 = 5;
    float load_5 = 6;
    float load_15 = 7;
  }
  // RAM
  message RAM {
    uint32 virt_total = 1;
    uint32 virt_available = 2;
    uint32 virt_used = 3;
    uint32 virt_free = 4;
    uint32 swap_total = 5;
    uint32 swap_used = 6;
    uint32 swap_free = 7;
    uint32 swap_percent = 8;
  }
  // Disk
  message Disk {
    repeated string partition_ids = 1;
    repeated string mount_point = 2;
    repeated string fstype = 3;
    repeated int32 total = 4;
    repeated int32 used = 5;
    repeated int32 free = 6;
    repeated float percent = 7;
    uint32 read_cnt = 8;
    uint32 write_cnt = 9;
    uint32 read_cytes = 10;
    uint32 write_bytes = 11;
    uint32 read_time = 12;
    uint32 write_time = 13;
  }
  // Battery
  message Battery {
    float percent = 1;
    uint32 secs_left = 2;
    bool power_plugged = 3;
  }
  // Session
  message Session {
    uint32 boot_time = 1;
    uint32 uptime = 2;
    repeated string users = 3;
    repeated string terminals = 4;
    repeated string hosts = 5;
    repeated uint32 started_time = 6;
  }

  CPU cpu = 4;
  RAM ram = 5;
  Disk disk = 6;
  Battery battery = 7;
  Session session = 8;


}
